### DISCUSSION TITLE
Different content of binaries\win64\ from different version

### DISCUSSION THREAD

**Discussion by Arinomo** [OpenModelica/OpenModelica#12767] [Q&A]
I export a model `foo.mo` using `v1.22.0-dev-39-g23fdc84c46 (64-bit)` and `foo_v2.mo` using `v1.22.1 (64-bit)`.

The first one has three files in `binaries\win64\ ` : `foo.lib` , `foo.dll` , and `foo_FMU.libs`

while the second one only has `binaries\win64\ ` :  `foo_v2.dll` , and `foo_v2_FMU.dll.a`

I wrote a small executable using exported FMU from OM, but it seems `foo_v2.fmu` triggered an error, because some dll are not found. 

why are the content of both fmus different? Do i need to modified some settings, so i get the same result as the first one?

Edit:
the `foo.fmu` exported using `v1.22.1` also contains only `.dll` and `.dll.a`

**Comment by AnHeuermann** [comment]
Hard to tell without an example and the exact way how you generated, but probably because we changed the way we compile FMUs at some point. See the [release notes](https://github.com/OpenModelica/OpenModelica/releases/tag/v1.22.0) on FMI support.
We switched from static libs compiled via Autotools (Makefiles) to (usually) dynamic libs compiled with CMake (other Makefiles).

If you encounter issue first try the latest nightly version of OpenModelica. Maybe we already fixed your issue. Otherwise it could be that dependencies like an ODE solver (libsundialscvode.dll or something like this) are now copied into the FMU to get a relocatable FMU. The issue is, that many importing tools are not loading additional DLLs correctly. One needs to specify a DLL directory before loading the foo_v2.dll.
We fixed that in FMI4C for OMSimulator: https://github.com/OpenModelica/OMSimulator-3rdParty/commit/ca418d7768c036ac15e9894d7f00d2118b3399a6
A workaround is to disable [--fmuRuntimeDepends](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/omchelptext.html#omcflag-fmuruntimedepends).

**Reply by Arinomo** [reply]
Thanks @AnHeuermann for the answer and sorry for a delayed reply. When exporting the model, I checked the `native` option and choose only to export `co-simulation` fmu. How can one export a statically linked fmu from OpenModelica, so it can be run on other machine, with for example FmPy? I know there is a cmake flag `BUILD_SHARED_LIBS` for recompiling the exported fmu, but is there any flag that i can set derectly on OMEdit? or in `cli` ?

Edit:
currently I can set the `--fmuCMakeBulid` to `false` and got the same result as from `v1.22.0-dev-39-g23fdc84c46`. FmPy on non modelica machine was able to simulate the fmu, so that's a work-around at the moment. I know `Makefile` build will be depreciated, but until then i have time to figure things out.

**Reply by AnHeuermann** [reply]
There is no CLI way to specify `BUILD_SHARED_LIBS=OFF`. And there are things we can't link statically into the FMU, like libpthreads.dll.

I'd recommend opening a ticket on your FMU importing tool and ask them to add `SetDllDirectoryA` before calling `LoadLibraryA`:

```C
BOOL success = SetDllDirectoryA(dllDirectory);
HINSTANCE dll = LoadLibraryA(dllPath);
```

**Comment by Arinomo** [comment]
So I installed `v1.24.0-dev-230-g20a061263e`, export following model using `gcc` and `clang` and try to simulate it using fmpy via jupyter notebook

```.mo
model LottkaVoltera
 parameter Real alpha=0.1 "Reproduction rate of prey";
 parameter Real beta=0.02 "Mortality rate of predator per prey";
 parameter Real gamma=0.4 "Mortality rate of predator";
 parameter Real delta=0.02 "Reproduction rate of predator per prey";
 parameter Real x0=10 "Initial prey population";
 parameter Real y0=10 "Initial predator population";
 output Real x(start=x0) "Prey population";
 output  Real y(start=y0) "Predator population";
initial equation
 x = x0;
 y = y0;
equation
 der(x) = x*(alpha-beta*y);
 der(y) = y*(delta*x-gamma);

end LottkaVoltera;
```
while the `gcc` version simulate without problem, i can't simulate the `clang` version. I assume its the problem of finding `libwinpthread-1.dll`.

where in `gcc` it got included in the fmu and linked
![image](https://github.com/user-attachments/assets/6456e1cd-57ae-423c-ba5a-570c25355344)

So basically, one need to used `gcc` in order to export `fmu` that most likely work on other machine? the `clang` version is in my case dependent to Meld

![image](https://github.com/user-attachments/assets/6132779d-d4aa-40cb-bb50-f5a7ae3efeef)

**Reply by Arinomo** [reply]
as reported in [#12065](https://github.com/OpenModelica/OpenModelica/issues/12065) and similar case on fmpy gihub [here](https://github.com/CATIA-Systems/FMPy/issues/407)
